include/forcing/GenericDataProvider.hpp

```diff
@@ -4,14 +4,26 @@
 #include "DataProvider.hpp"
 #include "DataProviderSelectors.hpp"
 
+#ifdef NETCDF_ACTIVE
+#include <memory>
+#include <netcdf>
+#endif
+
 namespace data_access
 {
     class GenericDataProvider : public DataProvider<double, CatchmentAggrDataSelector>
     {
         public:
 
+  #ifdef NETCDF_ACTIVE
+        /** Retrun the shared_ptr to the NcFile */
+        virtual std::shared_ptr<netCDF::NcFile> get_nc_file() {
+            return 0;
```

Comment:
If I am understanding this correctly, this will return `0` as a `shared_ptr<netCDF::NcFile>`... At a minimum, this should return `nullptr` instead... and as mentioned below, if you call `->close()` on this later it will crash.

Reply:
Yes, it should return a nullptr.